### PR TITLE
Enhance/#7706 - Update Enhanced Measurement Switch Functionality

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #7706 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/settings/SettingsEnhancedMeasurementSwitch.test.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ describe( 'SettingsEnhancedMeasurementSwitch', () => {
 			] );
 	} );
 
-	it( 'should render with the switch defaulting to the on position when enhanced measurement is enabled for the web data stream', async () => {
+	it( 'should render with the switch defaulting to the on position when enhanced measurement is disabled for the web data stream', async () => {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/settings/SettingsEnhancedMeasurementSwitch.test.js:114]: I don't think this change to the description needs to be made, as the test is still exercising the case where enhanced measurement is enabled for the stream, due to the call to `setEnhancedMeasurementStreamEnabled()` at the top of the test.

```suggestion
	it( 'should render with the switch defaulting to the on position when enhanced measurement is enabled for the web data stream', async () => {
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Nice work so far @hussain-t, there are just a few fairly minor points to address. Please take a look.

File: assets/js/modules/analytics-4/components/setup/SetupEnhancedMeasurementSwitch.js
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/setup/SetupEnhancedMeasurementSwitch.js:1]: Fixed ✅

File: assets/js/modules/analytics-4/components/settings/SettingsEnhancedMeasurementSwitch.test.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsEnhancedMeasurementSwitch.test.js:114]: Good catch 👍

File: assets/js/modules/analytics-4/datastore/enhanced-measurement.js [hunk 2]
```diff
@@ -429,6 +429,36 @@ const baseSelectors = {
...
+				state.enhancedMeasurement[ propertyID ]?.[ webDataStreamID ] ||
+				{};
+
+			// If the saved settings are not loaded yet, call the `getEnhancedMeasurementSettings` selector
+			// to invoke its resolver and load the saved settings.
+			if ( savedSettings === undefined ) {
+				select( MODULES_ANALYTICS_4 ).getEnhancedMeasurementSettings(
+					propertyID,
+					webDataStreamID
+				);
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/datastore/enhanced-measurement.js]: Updated ✅

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**